components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -859,93 +1418,56 @@ private static boolean isDefaultPoliciesExist(Connection connection) throws SQLE
-                        statement.setLong(3, ((RequestCountLimit) limit).getRequestCount());
-                        statement.setString(4, "");
-                    }
-                }
-                statement.setLong(5, pipeline.getQuotaPolicy().getLimit().getUnitTime());
-                statement.setString(6, pipeline.getQuotaPolicy().getLimit().getTimeUnit());
-                statement.execute();
-                List<Integer> conditionGroupIds = getConditionGroupIDs(connection, uuid);
-                int conID = conditionGroupIds.get(conditionGroupIds.size() - 1); //getting the last condition group id
-                List<Condition> conditionList = pipeline.getConditions();
-                for (Condition condition : conditionList) {
-                    if (condition instanceof IPCondition) {
-                        addIPCondition(connection, (IPCondition) condition, conID);
-                    } else if (condition instanceof HeaderCondition) {
-                        addHeaderCondition(connection, (HeaderCondition) condition, conID);
-                    } else if (condition instanceof JWTClaimsCondition) {
-                        addJWTClaimCondition(connection, (JWTClaimsCondition) condition, conID);
-                    } else if (condition instanceof QueryParameterCondition) {
-                        addParamCondition(connection, (QueryParameterCondition) condition, conID);
+                setDefaultThrottlePolicyDetailsPreparedStmt(limit, statement);
+                statement.executeUpdate();
+                ResultSet rs = statement.getGeneratedKeys();
+                if (rs.next()) {
+                    // get the auto increment id
+                    int conditionId = rs.getInt(1);
+                    List<Condition> conditionList = pipeline.getConditions();
+                    for (Condition condition : conditionList) {
+                        condition.populateDataInPreparedStatement(statement);
+                        if (PolicyConstants.IP_CONDITION_TYPE.equals(condition.getType()) ||
+                                PolicyConstants.IP_SPECIFIC_TYPE.equals(condition.getType()) ||
+                                PolicyConstants.IP_RANGE_TYPE.equals(condition.getType())) {
+                            addIPCondition(connection, condition, conditionId);
+                        } else if (PolicyConstants.HEADER_CONDITION_TYPE.equals(condition.getType())) {
+                            addHeaderCondition(connection, condition, conditionId);
+                        } else if (PolicyConstants.JWT_CLAIMS_CONDITION_TYPE.equals(condition.getType())) {
+                            addJWTClaimCondition(connection, condition, conditionId);
+                        } else if (PolicyConstants.QUERY_PARAMS_CONDITION_TYPE.equals(condition.getType())) {
+                            addParamCondition(connection, condition, conditionId);
+                        } else {
```

Comment:
If we do that, we will have to move the relevant SQL statements and DB related logic to the Condition concrete implementation as well, which IMHO goes against having all details of the DAO layer in the respective DAO class.